src/Datadog.Trace.ClrProfiler.Managed.Core/TypeExtensions.cs

```diff
@@ -26,6 +26,31 @@ internal static class TypeExtensions
             return null;
         }
 
+        public static System.Type GetInstrumentedType(
+            this object runtimeObject,
+            string instrumentedNamespace,
+            string instrumentedTypeName)
+        {
+            if (runtimeObject == null)
+            {
+                return null;
+            }
+
+            var currentType = runtimeObject.GetType();
+
+            while (currentType != null)
+            {
+                if (currentType.Namespace == instrumentedNamespace && currentType.Name == instrumentedTypeName)
```

Comment:
Would comparing the type name first before the namespace be a little faster in the common case? This is probably very tiny optimization, but also zero cost.

Two reasons: types can be in the same namespace, which in the current code means two string comparisons instead of one. Additionally, namespaces tend to be longer and start with a common substring. For example, when comparing `System.Data.SqlClient.SqlCommand` and `System.Data.Common.DbCommand`, the runtime can determine they are not equal as soon as it looks at the first character in the type names (`S` vs `D`), but it takes a little longer to compare the namespaces because they both start with `System.Data.`.

Reply:
Sounds reasonable, done

Reply:
Did this get implemented? It still looks like the namespace is tested first

Reply:
I flipped them in 8c39f42d